tests/utils.py

```diff
@@ -50,3 +52,57 @@ def generate_regression_model(features_shape, output_shape=1):
 def almost_equal(arr1, arr2, epsilon=1e-6):
     """Ensure two array are almost equal at an epsilon"""
     return np.sum(np.abs(arr1 - arr2)) < epsilon
+
+def generate_dataset(n_sample: int=100):
+    np.random.seed(0)
+
+    # dataset parameters
+    features_coef = np.array([1, -2, -3, 4, 0, 0, 0, 0])
+    nb_samples = 10000
+    nb_features = len(features_coef)
+
+    # create dataset
+    dataset = np.random.normal(4, 1, (nb_samples, nb_features))
+    noise = np.random.normal(0, 1, nb_samples)
+    target = dataset.dot(features_coef) + noise
+
+    return dataset, target
+
+def generate_linear_model(
+        X_train: np.array,
+        y_train: np.array,
+        library: str="keras"
+):
+    if library == "sklearn":
+        sk_linear = LinearRegression().fit(X_train, y_train)
+
+        # Create the wrapper class
+        class Wrapper:
+            # The init method is necessary for every class
+            def __init__(self, model):
+                self.model = model
+
+            # The call method calls the predict method
+            def __call__(self, inputs):
+                pred = self.model.predict(inputs)
+                return pred
+
+        sk_model = Wrapper(sk_linear)
+        return sk_model
+
+    elif library == "keras":
+        RANDOM_SEED = 1
```

Comment:
Seeding in tensorflow is a bit more complex, I've given up trying to get fairly reproducible results for tests -- the non-determinism of some ops doesn't help ^^

https://www.tensorflow.org/api_docs/python/tf/random/set_seed
https://github.com/tensorflow/tensorflow/issues/2732